crates/iota-ledger/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::vec;
+
+mod transport;
+use transport::{Transport, TransportTypes, create_transport};
+
+pub use crate::api::errors::LedgerError;
+mod api;
+use iota_types::{
+    base_types::IotaAddress,
+    crypto::{Ed25519IotaSignature, Signature, SignatureScheme, ToFromBytes},
+    object::Object,
+};
+use serde::Serialize;
+use shared_crypto::intent::{Intent, IntentMessage};
+
+use crate::api::{bolos, exit, get_public_key, sign_transaction};
+pub use crate::api::{get_public_key::PublicKeyResult, get_version::Version};
+
+/// Get Ledger by transport_type
+fn get_ledger_by_type(transport_type: TransportTypes) -> Result<Ledger, LedgerError> {
+    let transport = create_transport(transport_type)?;
+    Ok(crate::Ledger::new(transport))
+}
+
+pub struct Ledger {
+    transport: Transport,
+}
+
+pub struct SignedTransaction {
+    pub signature: Signature,
+    pub address: IotaAddress,
+}
+
+impl Ledger {
+    pub fn new_with_default() -> Result<Ledger, LedgerError> {
+        if std::env::var("LEDGER_SIMULATOR").is_ok() {
+            get_ledger_by_type(TransportTypes::TCP)
+        } else {
+            get_ledger_by_type(TransportTypes::NativeHID)
+        }
+    }
+
+    pub fn new_with_native_hid() -> Result<Ledger, LedgerError> {
+        get_ledger_by_type(TransportTypes::NativeHID)
+    }
+
+    pub fn new_with_simulator() -> Result<Ledger, LedgerError> {
+        get_ledger_by_type(TransportTypes::TCP)
+    }
+
+    fn new(transport: Transport) -> Self {
+        Ledger { transport }
+    }
+
+    /// Get currently opened app
+    /// If "BOLOS" is returned, the dashboard is open
+    pub fn is_app_open(&self) -> Result<bool, LedgerError> {
+        let app = bolos::app_get_name::exec(&self.transport)?;
+        Ok(app.app == "IOTA")
+    }
```

Comment:
Not sure if this is just an issue with the simulator and flex, but there it returns false, because the response looks like this:
```
Current app: Response { app: "app", version: "1.33.7" } 
```

Reply:
Yes, makes no sense in the simulator because there is no BOLOS running on that, only a single app instance